### ISSUE TITLE
QDrant startup takes too long, with ~50k vectors in 1 collection.

### ISSUE THREAD

**Issue by wtlyu** [qdrant/qdrant#3935]
## Current Behavior

Hi everyone. I'm facing a problem after restart qdrant in docker.

I have insert  ~50k vectors in 1 collection. everything goes ok. But If I restart the qdrant docker or copy storage dir and start a new docker on it. It takes too long to start, and before that every api is unreachable. (more than 1min to 1%)

## Steps to Reproduce
1. Run docker like this
```
docker run --name qdrant_test \
   -v "$(pwd)/data/qdrant_test:/qdrant/storage" \
   -v "$(pwd)/config/qdrant.yaml:/qdrant/config/production.yaml" \
   -p 8089:6333 \
   qdrant/qdrant
```
3. add vectors with 1 collection.
4. restart the docker, or copy the storage dir and start a new docker on it.
5. slow on Recovering collection, like :

```
               _
  __ _  __| |_ __ __ _ _ __ | |_
 / _` |/ _` | '__/ _` | '_ \| __|
| (_| | (_| | | | (_| | | | | |_
 \__, |\__,_|_|  \__,_|_| |_|\__|
    |_|

Version: 1.8.3, build: acb04d5f
Access web UI at http://localhost:6333/dashboard

2024-03-29T06:19:21.579150Z  INFO storage::content_manager::consensus::persistent: Loading raft state from ./storage/raft_state.json
2024-03-29T06:19:21.585233Z  INFO storage::content_manager::toc: Loading collection: jingask
2024-03-29T06:19:21.786770Z  INFO collection::shards::local_shard: Recovering collection jingask: 0/12726 (0%)
2024-03-29T06:20:21.965136Z  INFO collection::shards::local_shard: 160/12726 (1%)
^@2024-03-29T06:21:22.264951Z  INFO collection::shards::local_shard: 321/12726 (2%)
^@2024-03-29T06:22:22.531533Z  INFO collection::shards::local_shard: 483/12726 (3%)
^@2024-03-29T06:23:22.877498Z  INFO collection::shards::local_shard: 647/12726 (5%)
^@2024-03-29T06:24:23.177317Z  INFO collection::shards::local_shard: 810/12726 (6%)
^@2024-03-29T06:25:23.420688Z  INFO collection::shards::local_shard: 974/12726 (7%)
```

## Expected Behavior
It seems to start up the server in hours. Is it normal to take so0 long?

## Context (Environment)
```
{
  "result": {
    "status": "green",
    "optimizer_status": "ok",
    "vectors_count": 51888,
    "indexed_vectors_count": 49388,
    "points_count": 51888,
    "segments_count": 4,
    "config": {
      "params": {
        "vectors": {
          "size": 1536,
          "distance": "Cosine"
        },
        "shard_number": 1,
        "replication_factor": 1,
        "write_consistency_factor": 1,
        "on_disk_payload": true
      },
      "hnsw_config": {
        "m": 16,
        "ef_construct": 100,
        "full_scan_threshold": 10000,
        "max_indexing_threads": 0,
        "on_disk": true
      },
      "optimizer_config": {
        "deleted_threshold": 0.2,
        "vacuum_min_vector_number": 1000,
        "default_segment_number": 0,
        "max_segment_size": null,
        "memmap_threshold": 20000,
        "indexing_threshold": 20000,
        "flush_interval_sec": 5,
        "max_optimization_threads": 2
      },
      "wal_config": {
        "wal_capacity_mb": 32,
        "wal_segments_ahead": 0
      },
      "quantization_config": null
    },
    "payload_schema": {
      "project_uuid": {
        "data_type": "text",
        "points": 0
      },
      "document_uuid": {
        "data_type": "text",
        "points": 0
      }
    }
  },
  "status": "ok",
  "time": 0.000027426
}
```

config:
```
storage:
  # If true - point's payload will not be stored in memory.
  # It will be read from the disk every time it is requested.
  # This setting saves RAM by (slightly) increasing the response time.
  # Note: those payload values that are involved in filtering and are indexed - remain in RAM.
  on_disk_payload: true

  performance:
    # Max total number of threads, which can be used for running optimization processes across all collections.
    # Note: Each optimization thread will also use `max_indexing_threads` for index building.
    # So total number of threads used for optimization will be `max_optimization_threads * max_indexing_threads`
    max_optimization_threads: 2

  optimizers:
    # Maximum size (in KiloBytes) of vectors to store in-memory per segment.
    # Segments larger than this threshold will be stored as read-only memmaped file.
    # To enable memmap storage, lower the threshold
    # Note: 1Kb = 1 vector of size 256
    # To explicitly disable mmap optimization, set to `0`.
    # If not set, will be disabled by default.
    memmap_threshold_kb: 20000

    # Max number of threads, which can be used for optimization per collection.
    # Note: Each optimization thread will also use `max_indexing_threads` for index building.
    # So total number of threads used for optimization will be `max_optimization_threads * max_indexing_threads`
    # If `max_optimization_threads = 0`, optimization will be disabled.
    max_optimization_threads: 2

  # Default parameters of HNSW Index. Could be overridden for each collection or named vector individually
  hnsw_index:
    # Store HNSW index on disk. If set to false, index will be stored in RAM. Default: false
    on_disk: true
```

**Comment by wtlyu** [comment]
btw, it's just my test. because I want to use it with more than hundreds of millions of vector... need days or years to start?

**Comment by timvisee** [comment]
This issue has been resolved as we've discussed it in Discord here: https://discord.com/channels/907569970500743200/1223105029506990121

There were two reasons for this:
- A lot of delete by filter requests have been made, these are known to be very slow.
- The payload index for that filter field was configured incorrectly, which is why it wasn't used.

Setting up the payload index correctly resolved the problem. :smile:
